### ISSUE TITLE
Better logging

### ISSUE THREAD

**Issue by atilag** [QISKit/qiskit-sdk-py#88]
## Welcome contributors!
So far, we've been using a flag called "silent" (you can search for it into the sources), that if set to true, will cause some information to be printed out to the standard output.
We want to change it! 
So if you are willing to contribute to this amazing project (I know you are!), I'd suggest you to start with something simple like this issue.
We want to change all the "silent" flag behavior with a more robust and standard logging solution based in the python logging facility. So basically these are the tasks we need to address:
* Incorporate standard python logging module
* Create a logging config file with decent default settings:
    * INFO level
    * Console output
    * Set "disable_exisiting_loggers: false"
    * format:  "%(asctime)s : %(name)s : %(levelname)s : %(message)s"
* Loggers should be at module level and use the `__name__` attribute:
     * `logger = logging.getLogger(__name__)`

* Remove "silent" flag everywhere and replace the print() statements with logger.debug() (we don't want to show messages by default)

Suggestions are welcome! :)

### Notes
* If you want to take this bug, let us know and we will assign you to it.
* You can open a PR to "master" branch with the word: "WIP" (Work In Progress) in front of the description, so we will know that you haven't finished and we can make parcial reviewings. We will eliminate the WIP prefix once the PR is ready to be merged.
* Make sure all tests pass! :)
 

## Current Behavior
We are using the "silent" variable to print out information via standard output in just some modules. We want to remove them all and use a common logging system.

**Comment by hellmersjl3ds** [comment]
What's the status of this?  I see a commit from @ewinston but it's not assigned to anyone.  

I'd be happy to work on it if nobody else is.

**Comment by ewinston** [comment]
Hi @hellmersjl3ds, 
Thanks for offering to have a look at this. In PR#108, referenced in this PR, the python logging facility was setup and covers most of the features requested in this enhancement. One thing I think that remains to be done, however, is that the facility has not been applied everywhere (there are still `print` statements for instance). If you could help make the logging uniform or suggest any other improvements that would be great. For extensive changes you could open a new issue for discussion or continue using this one.

**Comment by lbello** [comment]
Currently, in 0.4, there are print calls in the following files:
 - `qiskit//tools/visualization.py`
 - `qiskit//tools/qi/qi.py`
 - `qiskit//tools/qi/pauli.py`
 - `qiskit//tools/apps/optimization.py`
 - `qiskit//unroll/_printerbackend.py` (probably makes sense here?)
 - `qiskit//qasm/_qasmlexer.py`
 - `qiskit//qasm/_node/_node.py`
 - `qiskit//qasm/_node/_qreg.py`
 - `qiskit//qasm/_node/_id.py`
 - `qiskit//qasm/_node/_real.py`
 - `qiskit//qasm/_node/_indexedid.py`
 - `qiskit//qasm/_node/_creg.py`
 - `qiskit//qasm/_node/_intnode.py`
 - `qiskit//qasm/_qasmparser.py`

This is a very nice opportunity for a first contribution. If anybody wants to work with it and need assistance, let me know! (<- @hellmersjl3ds)

**Comment by atilag** [comment]
> Not every `print` call in the code should be replaced by `log`. 

The problem of using `print()`statements, is that it forces the user to dump something to the  stdout (usually the monitor), but this is not true for many other uses cases that printing to stdout doesn't make any sense like:
* Graphics User Interfaces (Windows apps, Mac apps, XWindows apps)
* Non-Attended devices
* *nix Daemons and Services
* ... and probably many others I'm not aware of...

We are building an SDK, so using `print()` is killing many use-cases we don't really want to kill.
So, basically, yes, we should replace al the `prints` in the code by `log` :)
Notice that our logger can dump to stdout like `print` always does, if the user requires to do so... not sure if we are exposing this functionality to the user at the moment, though.

**Comment by lbello** [comment]
Some of the prints I mentioned are messages for the exception, for example.

**Comment by hellmersjl3ds** [comment]
Hey guys, unfortunately I'm busy with another project for the next 5 weeks or so, so I won't be able to get to this before then.

**Comment by atilag** [comment]
Ok, so after some PRs... we have included the logging system and we have removed all `silent` flags.
Still need to remove all presence of print() in the code, but we will create another issue to address this specific issue.
I'm going to close this Issue.
